### PR TITLE
Verify Tutorial 5: Common Types and Functions

### PR DESCRIPTION
**Pull request by barriebyron**
This PR presents edits for style and syntax.

No substantive changes were made. This tutorial is not really a set of guided steps, instead it's more like Tutorial 2 that presents code samples with technical architecture and mechanics. 

Ready for review, see my questions, thank yoU!

Closes https://app.zenhub.com/workspaces/zkapps-product-eng-6130fedb3b0fc600123d8796/issues/gh/o1-labs/docs2/300

### CODE REVIEW

File: docs/zkapps/tutorials/05-common-types-and-functions.mdx [hunk 1]
```diff
@@ -24,35 +28,36 @@ zkApp programmability is not yet available on the Mina Mainnet. You can get star
...
 
-In previous tutorials, we've seen how to deploy smart contracts to the network, and we've interacted with them from both a React UI and NodeJS.
+All types are defined in the [SnarkyJS / Modules / Types](/zkapps/snarkyjs-reference/modules/Types) API reference documentation.
 
-In this tutorial, we will talk about more types that are useful when building with SnarkyJS, so you can build more applications. So far, we've mostly been using the `Field` type. SnarkyJS provides other higher-order types built from Fields, that will be useful for your development.
+The [example project](https://github.com/o1-labs/docs2/tree/main/examples/zkapps/05-common-types-and-functions/src) includes a [main.ts](https://github.com/o1-labs/docs2/blob/main/examples/zkapps/05-common-types-and-functions/src/main.ts) file that shows all of the concepts presented in this tutorial, along with smart contracts showing more advanced usage of some of the concepts, particularly Merkle Trees.
```
Comment (barriebyron) [inline docs/zkapps/tutorials/05-common-types-and-functions.mdx:40]: are all concepts in the main.ts file?

File: docs/zkapps/tutorials/05-common-types-and-functions.mdx [hunk 2]
```diff
@@ -263,30 +271,30 @@ inputSumAbs: 5
...
+You can use [Merkle trees](../snarkyjs-reference/classes/MerkleTree) to manage large amounts of data within a circuit. The power of Merkle trees is demonstrated in the [05-common-types-and-functions/src](https://github.com/o1-labs/docs2/tree/main/examples/zkapps/05-common-types-and-functions/src) reference project for this tutorial. See the [BasicMerkleTreeContract.ts](https://github.com/o1-labs/docs2/blob/main/examples/zkapps/05-common-types-and-functions/src/BasicMerkleTreeContract.ts) contract and [main.ts](https://github.com/o1-labs/docs2/blob/main/examples/zkapps/05-common-types-and-functions/src/main.ts).
```
Comment (barriebyron) [inline docs/zkapps/tutorials/05-common-types-and-functions.mdx:295]: what does `main.ts` do with Merkle? we can describe

File: docs/zkapps/tutorials/05-common-types-and-functions.mdx [hunk 3]
```diff
@@ -341,7 +351,7 @@ Here is a simple example of a contract that stores the root of a merkle tree, wh
   }
 ```
 
-And code to interact with it:
+The code to interact with the smart contract:
```
Comment (barriebyron) [inline docs/zkapps/tutorials/05-common-types-and-functions.mdx:354]: by "it" do we mean "smart contract"?

File: docs/zkapps/tutorials/05-common-types-and-functions.mdx [hunk 4]
```diff
@@ -263,30 +271,30 @@ inputSumAbs: 5
...
 
-Note that when using `Provable.if`, like in a JS ternary, both branches are executed. Unlike normal programming, because SnarkyJS under the hood is creating a zk circuit, there is no primitive for if statements where only one branch is executed.
+Both branches are executed when using `Provable.if`, like in a JavaScript ternary. Because SnarkyJS is creating a zk circuit, there is no primitive for `if` statements where only one branch is executed.
 
 ## Assertions and Constraints
 
 SnarkyJS functions are compiled to generate circuits.
 
-When a transaction is proven in SnarkyJS, SnarkyJS is proving that the program logic is computed according to the written program, and all assertions are holding true.
+When a transaction is proven in SnarkyJS, the proof is that the program logic is computed according to the written program, and all assertions are holding true.
```
Comment (barriebyron) [inline docs/zkapps/tutorials/05-common-types-and-functions.mdx:280]: With this rewrite, we avoid having SnarkyJS do things.
Anthropomorphism guidance (don't do it) https://developers.google.com/style/anthropomorphism 

what other language can we use?

File: docs/zkapps/tutorials/05-common-types-and-functions.mdx [hunk 5]
```diff
@@ -401,13 +411,15 @@ console.log(
...
+
+This complete example is in the [project directory](https://github.com/o1-labs/docs2/tree/main/examples/zkapps/05-common-types-and-functions/src).
 
-You can find this complete example in the [project directory](https://github.com/o1-labs/docs2/tree/main/examples/zkapps/05-common-types-and-functions/src), as well as a more advanced example `LedgerContract`, which implements a basic ledger of tokens, including checks that the sender has signed their transaction and that the amount the sender has sent matches the amount the receiver receives.
+A more advanced example `LedgerContract` implements a basic ledger of tokens, including checks that the sender has signed their transaction and that the amount the sender has sent matches the amount the receiver receives.
 
 ## Merkle Map
 
-Lastly, let's go over an example using [merkle maps](../snarkyjs-reference/classes/MerkleMap). These allow us to implement key value stores.
+See the API reference documentation for [Merkle maps](../snarkyjs-reference/classes/MerkleMap). These allow us to implement key value stores.
```
Comment (barriebyron) [inline docs/zkapps/tutorials/05-common-types-and-functions.mdx:422]:
```suggestion
See the API reference documentation for the [MerkleMap](../snarkyjs-reference/classes/MerkleMap) class you can use to implement key-value stores.
```

File: docs/zkapps/tutorials/05-common-types-and-functions.mdx [hunk 6]
```diff
@@ -2,7 +2,7 @@
 title: 'Tutorial 5: Common Types and Functions'
 hide_title: true
 sidebar_label: 'Tutorial 5: Common Types and Functions'
-description: Higher-order types are built from Fields and are useful for your development. Learn how to create custom types.
+description: Higher-order types are built from Fields. Create custom types to build your own strings. Create compound data types with the Struct.
```
Comment (Trivo25) [inline docs/zkapps/tutorials/05-common-types-and-functions.mdx:5]: > Create custom types to build your own strings

I think we should slightly change this, since this tutorial isn't only about strings, its more about common data types, functions and how to compose and use them

File: docs/zkapps/tutorials/05-common-types-and-functions.mdx (hunk 1 above)
Comment (Trivo25) [reply docs/zkapps/tutorials/05-common-types-and-functions.mdx:40]: yup!

File: docs/zkapps/tutorials/05-common-types-and-functions.mdx [hunk 7]
```diff
@@ -98,20 +103,20 @@ Fields in char1: 1
...
-4 more advanced types are:
+Four advanced types are:
 
 - [CircuitString](../snarkyjs-reference/classes/CircuitString)
 - [PrivateKey](../snarkyjs-reference/classes/PrivateKey)
 - [PublicKey](../snarkyjs-reference/classes/Types.PublicKey)
 - [Signature](../snarkyjs-reference/classes/Signature)
 
-All arguments passed into smart contracts need to be arguments SnarkyJS can understand. This means that we can't just pass normal strings - we need to pass in strings that have been wrapped to be compatible with circuits, which is what `Struct` accomplishes.
+All arguments passed into smart contracts must be arguments SnarkyJS can consume. You cannot pass normal strings. Instead, you must pass in strings that are wrapped to be compatible with circuits. This is accomplished with `Struct`.
```
Comment (Trivo25) [inline docs/zkapps/tutorials/05-common-types-and-functions.mdx:113]: > You cannot pass normal strings

I would rephrase this to 

> You cannot pass normal JavaScript-native types, such as strings, number, boolean or bigint

so this is more general and doesn't just focus on strings

Comment (Trivo25) [reply docs/zkapps/tutorials/05-common-types-and-functions.mdx:113]: same here

>strings that are wrapped to be compatible with circuits

File: docs/zkapps/tutorials/05-common-types-and-functions.mdx (hunk 2 above)
Comment (Trivo25) [reply docs/zkapps/tutorials/05-common-types-and-functions.mdx:295]: In `main.ts` we demonstrate how contracts interact with Merkle trees and how to construct them

File: docs/zkapps/tutorials/05-common-types-and-functions.mdx [hunk 8]
```diff
@@ -296,18 +304,20 @@ import {
...
 ```ts
 const height = 20;
 const tree = new MerkleTree(height);
 ```
 
-The height variable determines how many leaves are available to your application. A height of 20 for example will lead to a tree with `2^(20-1)`, or 524,288 leaves.
+The height variable determines how many leaves are available to the application. For example, a height of 20 leads to a tree with `2^(20-1)`, or 524,288 leaves.
 
-Merkle trees in smart contracts are stored as the hash of the merkle tree's root. Smart contract methods which update the merkle root can take a "witness" of the change as an argument, which represents the merkle path to the data for which inclusion is being proved.
+Merkle trees in smart contracts are stored as the hash of the Merkle tree's root. Smart contract methods that update the Merkle root can take a _witness_ of the change as an argument. The witness represents the Merkle path to the data for which inclusion is being proved.
```
[the remark this answers is not in the document]
Comment (Trivo25) [reply docs/zkapps/tutorials/05-common-types-and-functions.mdx:316]: yes! Would be good to link it

**Review by Trivo25 — APPROVED** [review]
This is great! Really like the changes, this will enable developers to more easily learn SnarkyJS!

File: docs/zkapps/tutorials/05-common-types-and-functions.mdx (hunk 2 above)
Comment (barriebyron) [inline docs/zkapps/tutorials/05-common-types-and-functions.mdx:295]:
```suggestion
You can use [Merkle trees](../snarkyjs-reference/classes/MerkleTree) to manage large amounts of data within a circuit. The power of Merkle trees is demonstrated in the [05-common-types-and-functions/src](https://github.com/o1-labs/docs2/tree/main/examples/zkapps/05-common-types-and-functions/src) reference project for this tutorial. See the [BasicMerkleTreeContract.ts](https://github.com/o1-labs/docs2/blob/main/examples/zkapps/05-common-types-and-functions/src/BasicMerkleTreeContract.ts) contract and [main.ts](https://github.com/o1-labs/docs2/blob/main/examples/zkapps/05-common-types-and-functions/src/main.ts) that demonstrates how contracts interact with Merkle trees and how to construct them.
```

File: docs/zkapps/tutorials/05-common-types-and-functions.mdx (hunk 8 above)
Comment (barriebyron) [inline docs/zkapps/tutorials/05-common-types-and-functions.mdx:316]:
```suggestion
Merkle trees in smart contracts are stored as the hash of the Merkle tree's root. Smart contract methods that update the Merkle root can take a _witness_ of the change as an argument. The [MerkleMapWitness](/zkapps/snarkyjs-reference/classes/MerkleMapWitness) represents the Merkle path to the data for which inclusion is being proved.
```
